rails/app/models/activity.rb

```diff
@@ -22,27 +22,17 @@ def student_only
       where('teacher_only' => false)
     end
   end
-  has_many :pages, :through => :sections
-  has_many :project_materials, :class_name => "Admin::ProjectMaterial", :as => :material, :dependent => :destroy
-  has_many :projects, :class_name => "Admin::Project", :through => :project_materials
 
-  # BASE_EMBEDDABLES is defined in config/initializers/embeddables.rb
-  # This block adds a has_many for each embeddable type to this model.
-  # TODO We don't want Embeddable::Iframe showing up in any menus, so inject it here. It's used by LARA.
-  (BASE_EMBEDDABLES + ["Embeddable::Iframe"]).each do |klass|
-      eval %!has_many :#{klass[/::(\w+)$/, 1].underscore.pluralize}, :class_name => '#{klass}',
-      :finder_sql => proc { "SELECT #{klass.constantize.table_name}.* FROM #{klass.constantize.table_name}
-      INNER JOIN page_elements ON #{klass.constantize.table_name}.id = page_elements.embeddable_id AND page_elements.embeddable_type = '#{klass}'
-      INNER JOIN pages ON page_elements.page_id = pages.id
-      INNER JOIN sections ON pages.section_id = sections.id
-      WHERE sections.activity_id = \#\{id\}" }!
+  has_many :pages, :through => :sections do
+    def student_only
```

Comment:
Is this new method still needed?

Reply:
Thanks -- no, this must have been one of the experiments we were working on that I fogot to back out.  I will remove it.

Reply:
edad8bdaf2281c62f044ea53ba75db802e406537 undoes this.